csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/ExternalLocationSink.qll

```diff
@@ -63,3 +64,68 @@ class CookieStorageSink extends ExternalLocationSink, RemoteFlowSink {
     )
   }
 }
+
+private predicate isFileWriteCall(Expr stream, Expr data) {
```

Comment:
Would it be possible to use Models as Data instead (maybe define a new sink kind if needed)?

Reply:
Possibly; however since we care about two different parameters (the stream and the written data) then we'd need to encode that information in the sink kind; similar to [regex use sinks](https://github.com/github/codeql/blob/main/java/ql/lib/ext/java.lang.model.yml#L24) in Java [are used](https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/regex/RegexFlowConfigs.qll#L21) by the ReDoS queries. Is this worthwhile?

The sources for this config could also more straigtforwardly be converted to MaD.

Reply:
If it add extra complexity just leave it as it is - unless you prefer to do otherwise.
If the sources are easily convertible to MaD then it is probably a good idea (this will be relevant for the high level configuration prototype for including *more* (all sources)) when running a query.